app/repositories/generator/DataGenerator.java

```diff
@@ -69,29 +66,41 @@ public static Room generateRoom() {
             room.setLength(random.nextDouble() * 100);
             room.setWidth(random.nextDouble() * 100);
 
-            //TODO: Set measurements
-
             return room;
         } catch(final Exception ex) {
             throw new DataGeneratorException("Failed generating a single room", ex);
         }
     }
 
-    public static List<MeasurementReadings> generateMeasurements(final int amount) {
+    public static List<MeasurementMetadata> generateMeasurements(final int amount) {
         try {
-            final List<MeasurementReadings> measurementReadings = new ArrayList<>();
+            final List<MeasurementMetadata> measurements = new ArrayList<>();
 
             for (int i = 0; i < amount; i++) {
-                measurementReadings.add(generateMeasurement());
+                measurements.add(generateMeasurementMetadata());
             }
 
-            return measurementReadings;
+            return measurements;
         }
         catch(final Exception ex) {
             throw new DataGeneratorException("Failed generating a batch of measurements", ex);
         }
     }
 
+    public static List<Reading> generateReadings(final int amount) {
+        try {
+            final List<Reading> readings = new ArrayList<>(amount);
+
+            for(int i = 0; i < amount; i++) {
+                readings.add(generateReading());
+            }
+
+            return readings;
+        } catch(final Exception ex) {
+            throw new DataGeneratorException("Failed generating a multiple readings", ex);
```

Comment:
Wrong sentence: "a multiple readings".
Better would be "Error while generating the readings". It's not clear if not some readings were created successfully...